// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Copyright 2015 and onwards Google, Inc.
// The SparrowhawkConfiguration stores information about the grammars, and rules
// to run.

syntax = "proto2";

package speech.sparrowhawk;

message SparrowhawkConfiguration {
  optional string tokenizer_grammar = 1;  // Tokenizer-classifier.

  optional string verbalizer_grammar = 2;

  // Regular expression for sentence boundary detector. This is a set
  // of possible end-of-sentence markers.
  optional string sentence_boundary_regexp = 3;

  // Optional file specifying tokens that end in a possible end-of-sentence
  // marker that should *not* usually induce an end-of-sentence decision
  // e.g. “Mr.”
  optional string sentence_boundary_exceptions_file = 4;

  // Optional file with SerializeSpec for verbalizer as a text proto. If the
  // the field is not set, we resort to protobuf serializer.
  optional string serialization_spec = 5;
}
